app/controllers/reauthentication_controller.rb

```diff
@@ -0,0 +1,10 @@
+class ReauthenticationController < ApplicationController
+  before_action :confirm_two_factor_authenticated
+
+  def create
+    user_session[:stored_location] = account_url
+    user_session[:context] = 'reauthentication'
+
+    redirect_to login_two_factor_options_path(reauthn: true)
+  end
```

Comment:
Could we move the reauthentication controller to under the accounts module. I think having a "reauthn" method in the accounts controller could work but would be less "clean" imo.

Reply:
I think it may be easier to maintain if the actions are near one another in the same file. Users will be hitting /account in the AccountsController and it currently only has one route currently, so I think having the two actions next to each other might be clearer too.

Reply:
Could we move the route into the AccountsController?

Reply:
> Could we move the route into the AccountsController?

How strongly do you feel about this? Curious what @mdiarra3 thinks, but personally I like it as implemented currently, vs. introducing some non-standard ad hoc controller action.

Reply:
Not that strongly. To me, since it is a route that is only used within the `AccountController#show`, and it redirects back to the `AccountController#show`, keeping it in the same file (with associated comments/documentation) would be clearer when for maintaining this going forward.

Reply:
Okay I think I can move it to accounts controller